subsys/bluetooth/host/keys.c

```diff
@@ -37,6 +39,36 @@ static struct bt_keys key_pool[CONFIG_BT_MAX_PAIRED];
 #if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
 static uint32_t aging_counter_val;
 static struct bt_keys *last_keys_updated;
+
+#if (CONFIG_BT_MAX_CONN > 1)
+static bool bt_key_is_not_in_use(uint8_t id)
+{
+	struct bt_conn *conn;
+	struct bt_keys *key;
+	bool not_in_use = true;
+	int i;
+
```

Comment:
To help with ridding the code of preprocessor directives, just remove the `#if (CONFIG_BT_MAX_CONN > 1)` on the top and put this in the beginning of `bt_key_is_not_in_use()`:
```
if (CONFIG_BT_MAX_CONN < 2) {
        return true;
}
```

Reply:
Done